src/mixins/filing-mixin.ts

```diff
@@ -173,13 +179,8 @@ export default class FilingMixin extends DateMixin {
         legalType: this.getEntityType
       },
       restoration: {
-        // FUTURE: update as needed
-        date: null,
-        type: RestorationTypes.FULL,
-        expiry: null
+        type: restorationType
       }
     }
```

Comment:
You may want to cast this anonymous object "as RestorationFilingIF" so any missing or mis-typed properties are identified.

Reply:
Hmm, the method already has a return type of RestorationFilingIF.  I'm not clear what you're asking.  Sorry, it's Friday afternoon, I may not be thinking clearly.
```
buildRestorationFiling (restorationType = RestorationTypes.FULL): RestorationFilingIF {
    return {
      header: {
        date: this.getCurrentDate,
        name: FilingTypes.RESTORATION
      },
      business: {
        foundingDate: this.dateToApi(this.getEntityFoundingDate),
        identifier: this.getIdentifier,
        legalName: this.entityName,
        legalType: this.getEntityType
      },
      restoration: {
        type: restorationType
      }
    }
  }
```

Reply:
The method is fine, but VS Code doesn't show errors in the anonymous object you're creating (as long as it's upward-compatible with the method return value). So it won't show missing fields. Try it.

Reply:
I believe this doesn't complain:
```
  private formatAddress (address: any): AddressIF {
    return {
      severin: true,
      addressCity: address.addressCity || '',
      addressCountry: address.addressCountry || '',
      addressRegion: address.addressRegion || '',
      deliveryInstructions: address.deliveryInstructions || '',
      postalCode: address.postalCode || '',
      streetAddress: address.streetAddress || '',
      streetAddressAdditional: address.streetAddressAdditional || ''
    }
  }
```

But this does:
```
  private formatAddress (address: any): AddressIF {
    return {
      severin: true,
      addressCity: address.addressCity || '',
      addressCountry: address.addressCountry || '',
      addressRegion: address.addressRegion || '',
      deliveryInstructions: address.deliveryInstructions || '',
      postalCode: address.postalCode || '',
      streetAddress: address.streetAddress || '',
      streetAddressAdditional: address.streetAddressAdditional || ''
    } as AddressIF
  }
```

Reply:
Ah, got it.  Pushing update shortly

Reply:
Yes, that works too :)